transformer_engine/jax/fp8.py

```diff
@@ -82,72 +83,67 @@ class FP8MetaPackage:
     A container that contains all required meta data for FP8
     """
 
+    NUM_OF_META: int = 3
+    INPUT_IDX: int = 0
+    WEIGHT_IDX: int = 1
+    GRAD_IDX: int = 2
+
     def __init__(
         self,
-        num_of_gemm: int,
-        fp8_max: jnp.ndarray,
-        amax: jnp.ndarray,
-        scale: jnp.ndarray,
-        scale_inv: jnp.ndarray,
+        i_amax: jnp.ndarray,
+        i_scale: jnp.ndarray,
+        w_amax: jnp.ndarray,
+        w_scale: jnp.ndarray,
+        g_amax: jnp.ndarray,
+        g_scale: jnp.ndarray,
     ) -> None:
-        total_num_of_meta = num_of_gemm * FP8Helper.NUM_META_PER_GEMM
-        self._num_of_gemm = num_of_gemm
-        assert fp8_max.shape[0] == total_num_of_meta
-        self._fp8_max = fp8_max
-        assert amax.shape[0] == total_num_of_meta
-        self._amax = amax
-        assert scale.shape[0] == total_num_of_meta
-        self._scale = scale
-        assert scale_inv.shape[0] == total_num_of_meta
-        self._scale_inv = scale_inv
 
-    @property
-    def num_of_gemm(self) -> int:
-        """
-        num_of_gemm of this package
-        """
-        return self._num_of_gemm
+        self._amax_list = [None] * FP8MetaPackage.NUM_OF_META
+        self._scale_list = [None] * FP8MetaPackage.NUM_OF_META
 
-    @property
-    def fp8_max(self) -> jnp.ndarray:
-        """
-        fp8_max of this package
-        """
-        return self._fp8_max
+        self._amax_list[FP8MetaPackage.INPUT_IDX] = i_amax
+        self._scale_list[FP8MetaPackage.INPUT_IDX] = i_scale
+        self._amax_list[FP8MetaPackage.WEIGHT_IDX] = w_amax
+        self._scale_list[FP8MetaPackage.WEIGHT_IDX] = w_scale
+        self._amax_list[FP8MetaPackage.GRAD_IDX] = g_amax
+        self._scale_list[FP8MetaPackage.GRAD_IDX] = g_scale
 
     @property
-    def amax(self) -> jnp.ndarray:
+    def amax_list(self) -> List[jnp.ndarray]:
```

Comment:
Do we need to add `_list` suffix? I think the type annotation is enough to represent it's data structure.

Reply:
Adding `_list` here is to distinguish the `amax` tensors and `amax` containers, then we could recognize what the var contains by its name. Like `amax_list` means a list of multiple amax tensors and `amax` means an amax tensor.
But I have no strong thought here, we could discuss offline to make a final call.